client/my-sites/plans-grid/lib/filter-unused-features-object.ts

```diff
@@ -13,18 +13,18 @@ const filterUnusedFeaturesObject = (
 	}
 
 	// Get all unique feature slugs in all gridPlans.
-	const uniqueFeaturesAvailable = Array.from(
-		new Set(
-			visibleGridPlans
-				.map( ( gridPlan ) =>
-					gridPlan.features.wpcomFeatures.map( ( feature ) => feature.getSlug() )
+	const uniqueFeaturesAvailable = new Set(
+		visibleGridPlans
+			.map( ( gridPlan ) =>
+				[ ...gridPlan.features.wpcomFeatures, ...gridPlan.features.jetpackFeatures ].map(
+					( feature ) => feature.getSlug()
 				)
-				.flat()
-		)
+			)
+			.flat()
 	);
 
 	return features.filter( ( feature ) => {
-		return uniqueFeaturesAvailable.includes( feature.getSlug() );
+		return uniqueFeaturesAvailable.has( feature.getSlug() );
```

Comment:
The core fix here was to include `gridPlan.features.jetpackFeatures` as part of the filtering.

However I noticed the conversion of `uniqueFeaturesAvailable` to an array appears to be redundant, so it's now left as a set.